src/composables/transactionList.ts

```diff
@@ -129,122 +131,46 @@ export function useTransactionList({ store }: IDefaultComposableOptions) {
     waitTransactionMined(address, transaction.hash);
   }
 
-  async function fetchTransactionsFromMiddleware(
-    transactionState: ITransactionsState,
-    address: Encoded.AccountAddress,
-    recent: boolean,
-    limit: number,
-  ) {
-    const url = (recent || transactionState.nextPageUrl === '')
-      ? `/v2/accounts/${address}/activities?limit=${limit}`
-      : transactionState.nextPageUrl!;
-
-    try {
-      const { data, next } = await fetchFromMiddlewareCamelCased(url);
-      if (!recent) {
-        setTransactionsNextPage(address, next);
-      }
-      return data || [];
-    } catch (error) {
-      return [];
-    }
-  }
-
-  async function fetchPendingTransactions(address: Encoded.AccountAddress) {
-    const sdk = await getAeSdk();
-
-    try {
-      const fetchedPendingTransaction = (
-        await sdk.api.getPendingAccountTransactionsByPubkey(address)
-      );
-      const transactionState = getAccountTransactionsState(address);
-
-      return JsonBig.parse(JsonBig.stringify(
-        fetchedPendingTransaction?.transactions || [],
-      ))
-        .filter(
-          (transaction: ITransaction) => !transactionState.pending[nodeNetworkId.value!]?.find(
-            (tx) => tx?.hash === transaction?.hash,
-          ),
-        ).map((transaction: ITransaction) => ({ ...transaction, pending: true }));
-    } catch (error) {
-      return [];
-    }
-  }
-
-  async function fetchTipWithdrawnTransactions(address: Encoded.AccountAddress, recent: boolean) {
-    try {
-      await getAeSdk();
-      const response = await fetchJson(
-        `${aeActiveNetworkSettings.value.backendUrl}/cache/events/?address=${address}&event=TipWithdrawn${recent ? '&limit=5' : ''}`,
-      );
-      if (response.message) {
-        return [];
-      }
-      // TODO prepare interface for response
-      const tipWithdrawnTransactions: ITransaction[] = (response as any[]).map(({
-        amount,
-        contract,
-        height,
-        data: { tx },
-        ...t
-      }) => ({
-        tx: {
-          ...tx,
-          address,
-          amount,
-          contractId: contract,
-          type: Tag[Tag.ContractCallTx],
-        },
-        ...t,
-        microTime: new Date(t.createdAt).getTime(),
-        blockHeight: height,
-        claim: true,
-      }));
-
-      return tipWithdrawnTransactions;
-    } catch (error) {
-      return [];
-    }
-  }
-
   async function fetchTransactions(
-    limit: number,
-    recent: boolean,
     address: Encoded.AccountAddress,
+    protocol?: Protocol,
   ): Promise<ITransaction[]> {
-    await getAeSdk(); // Ensure the `nodeNetworkId` is established
-
+    await getAeSdk();
```

Comment:
Why did you remove the comment? It explained why we wait for the SDK without actually using it.